src/DateInput.tsx

```diff
@@ -358,3 +360,75 @@ export function DateInputFieldWithLabel (
     </>
   )
 }
+
+// Type-in Date Input
+
+const dateMask = [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]
+
+export function DateInputTypeIn ({
+  name,
+  hasError,
+  onChange,
+  onBlur
+}: {
+  name: string
+  hasError?: boolean
+  onChange(val: string): void
+  onBlur(): void
+}) {
+  return (
+    <MaskedInput
+      name={name}
+      mask={dateMask}
+      id='date'
+      placeholder='mm/dd/yyyy'
+      type='text'
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+        onChange(e.target.value)
+      }}
+      onBlur={onBlur}
+      render={(inputRef, props) => (
+        <Input name={name} ref={inputRef} hasError={hasError} {...props} />
+      )}
+    />
+  )
+}
+
+export function DateInputTypeInField ({ name }: { name: string }) {
+  const validate = React.useCallback((value: string) => {
+    if (!value) {
+      return 'Required'
+    }
+
+    const moment = require('moment')
+    const [month, day, year] = value.split('/').map(v => parseInt(v))
+
+    if (year < 1900) {
+      return 'The date you entered is not a valid date.'
+    }
+
+    const dateIsValid = moment([year, month - 1, day]).isValid()
+    if (!dateIsValid) {
+      return 'The date you entered is not a valid date.'
+    }
+
+    return
+  }, [])
+
+  return (
+    <Field name={name} validate={validate}>
+      {({ form }: FieldProps) => {
+        const hasError = Boolean(get(form, ['errors', name]))
+
+        return (
+          <DateInputTypeIn
+            name={name}
+            hasError={hasError}
+            onChange={date => form.setFieldValue(name, date)}
+            onBlur={() => form.setFieldTouched(name)}
+          />
+        )
+      }}
```

Comment:
Did you notice how in SSNInput that Formik provides a `field` prop here that contains logic for onChange and stuff? Probably best to use that instead of handling it yourself.

To do that, you'll need to update `onChange` to match a normal `onChange` handler, instead of a custom `onChange(val: string)`.

Reply:
Thanks for pointing this out! I made this fix, and in doing so, tried to better type these components as well. I _think_ we can reuse the `InputProp` type here, because the underlying component is our Input component. Lmk if this sounds right to you!